File: homework/homework1/code_matlab_discarded/p1/p1a.m
clear;clc;

origin_grain = imread('../../origin_images/grain.tif');
gray_num = p1_count_gray_num(origin_grain);

% plot the histogram
figure(1);
bar(0:255, gray_num);
xlabel('gray level');
ylabel('number of pixels');
title('histogram of grain.tif');

File: homework/homework1/code/p1/p1a.py
import matplotlib.pyplot as plt
from p1_count_gray_num import count_gray_num

origin_grain = plt.imread('./origin_images/grain.tif')
gray_num = count_gray_num(origin_grain)


fig, ax = plt.subplots(1, 2)
        
# show the origin image
ax[0].imshow(origin_grain, cmap='gray')
ax[0].set_title('origin image')

# show the histogram of grain.tif
ax[1].bar(range(256), gray_num)
ax[1].set_xlabel('gray level')
ax[1].set_ylabel('number of pixels')
ax[1].set_title('histogram of grain.tif')
plt.show()